ssl-config/src/main/java/com/palantir/remoting/ssl/KeyStores.java

```diff
@@ -26,40 +29,97 @@
 import java.nio.file.Path;
 import java.security.GeneralSecurityException;
 import java.security.Key;
+import java.security.KeyFactory;
 import java.security.KeyStore;
+import java.security.PrivateKey;
 import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
 import java.security.cert.CertificateFactory;
+import java.security.spec.RSAPrivateKeySpec;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 final class KeyStores {
 
+    /**
+     * Pattern that matches a single RSA key in a PEM file. Has a capture group that captures
+     * the content of the key (everything that occurs between the header and footer).
+     */
+    private static final Pattern KEY_PATTERN = Pattern.compile(
+            "-----BEGIN RSA PRIVATE KEY-----(.+)?-----END RSA PRIVATE KEY-----", Pattern.DOTALL);
```

Comment:
Ah, got it. Yeah right now the code strips all newlines so didn't have that, but looks like newline after header and before footer is mandated by the format, so should be able to bake into the regex. Will do.